ops/model.py

```diff
@@ -3792,6 +3792,40 @@ def _ensure_loaded(self):
         assert self._notice.key == self.key
 
 
+class LazyCheckInfo:
+    """Provide lazily-loaded access to a Pebble check's info.
+
+    The attributes provided by this class are the same as those of
+    :class:`ops.pebble.CheckInfo`, however, the notice details are only fetched
+    from Pebble if necessary (and cached on the instance).
+    """
+
+    name: str
+    level: Optional[Union[pebble.CheckLevel, str]]
+    status: Union[pebble.CheckStatus, str]
```

Comment:
What is the reason for union types here? Is it because other, custom values are possible?

Reply:
Here it's because this is really just a wrapper over `CheckInfo` so it needs to have the same types.

In the real `CheckInfo`, I believe it's so that if Pebble gains additional status or level values, ops will keep working. @benhoyt might have more to say on this?

Reply:
Yeah, that's right -- if Pebble gains additional values but they haven't been added to Ops yet, we won't get exceptions. However, in retrospect I'm not sure it was a good choice. The "enum or str" type is very awkward.

Reply:
I think `StrEnum|str` is a lot less awkward to deal with, so we could switch to that once 3.11 can be a minimum.

Reply:
I guess this can't helped, resolving.

Reply:
FWIW, my preference in a situation like this would be:

```py
status: CheckStatus|None
raw_status: str
```

That way, most callers use `.status`.
When a charmer tries to use the new pebble feature, they are nudged to upgrade ops by the static type checker.
A determined, enlightened caller who depends on unreleased pebble functionality can still get their custom value.


Another way would be to rely on Literals instead of Enums.